Sources/LanguageServerProtocol/SupportTypes/FileSystemWatcher.swift

```diff
@@ -0,0 +1,43 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Defines a watcher interested in specific file system change events.
+public struct FileSystemWatcher: Codable, Hashable {
+  /// The glob pattern to watch.
+  public var globPattern: String
+
+  /// The kind of events of interest. If omitted it defaults to
+  /// WatchKind.create | WatchKind.change | WatchKind.delete.
+  public var kind: WatchKind?
+
+  public init(globPattern: String, kind: WatchKind? = nil) {
+    self.globPattern = globPattern
+    self.kind = kind
+  }
+}
+
+/// The type of file event a watcher is interested in.
+///
+/// In LSP, this is an integer, so we don't use a closed set.
+
+// fixme: optionset?
```

Comment:
LMK if you think it makes sense to swap over

Reply:
Yeah, OptionSet makes sense to me 👍

Reply:
Done